src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class BuildWorkspace : Workspace
+    {
+        public BuildWorkspace(ProjectReaderSettings settings) : base(settings) { }
+
+        /// <summary>
+        /// Create an empty <see cref="WorkspaceContext" /> using the default <see cref="ProjectReaderSettings" />, with the specified Version Suffix
+        /// </summary>
+        /// <param name="versionSuffix">The suffix to use to replace any '-*' snapshot tokens in Project versions.</param>
+        /// <returns></returns>
+        public static BuildWorkspace Create(string versionSuffix)
+        {
+            var settings = ProjectReaderSettings.ReadFromEnvironment();
+            if (!string.IsNullOrEmpty(versionSuffix))
```

Comment:
I could, and probably will. But most callers take a command line arg and send it in here. So rather than force them to do the check and call a different overload, I can do the check here.